### PR TITLE
Solana Versioned Transactions

### LINKED ISSUE #697: Support Solana Versioned Transactions
**Describe the bug**
Current setup does not support Solana's new versioned transactions. https://docs.solana.com/developing/versioned-transactions

Some areas to fix
- add maxSupportedTransactionVersion field to get get_transaction functions
```

      transaction_receipt = solana_api.api.get_transaction(
                Signature.from_string(transaction_digest),
                max_supported_transaction_version=100
            ) 
```
            
- signing will need to be able to handle as well, here is some sample code to add nonce, and sign a versionedtx 

```
       recent_blockhash = self.ledger_api.api.get_latest_blockhash().value.blockhash
        txn['message'][1]['recentBlockhash'] = json.loads(recent_blockhash.to_json())
        msg = MessageV0.from_json(json.dumps(txn['message'][1]))
        signed_transaction = VersionedTransaction(msg,[payer.entity])

```
- function to convert dict tx to versiontx and back. There is a bug in the solders lib that doesnt allow direct from_json(). But you can pull out the versioned msg (txn['message'][1]) and load it into a new versioned tx

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

fixes #697 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py [hunk 1]
```diff
@@ -64,6 +64,9 @@
 from aea.helpers.base import try_decorator
 
 
+DEFAULT_MAX_SUPPORTED_TRANSACTION_VERSION = 100
```
Comment (angrybayblade) [inline plugins/aea-ledger-solana/aea_ledger_solana/solana.py:67]: @8ball030 @Dassy23 Is this correct for default max supported transaction version?

Comment (Dassy23) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:67]: No I think it should be 0
Refer to https://solanacookbook.com/guides/versioned-transactions.html#rpc-changes

Comment (angrybayblade) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py]: Addressed https://github.com/valory-xyz/open-aea/pull/699/commits/e0bc608634c69c88bbb78372133ec9d458266d51

File: plugins/aea-ledger-solana/aea_ledger_solana/crypto.py [hunk 2]
```diff
@@ -171,14 +172,22 @@ def _sign_tx_legacy(
...
         recent_hash = Hash.from_string(transaction["recentBlockhash"])
-        stxn = Transaction.from_json(jsonTx)
+        if isinstance(transaction["message"], list):
+            transaction["message"][1]["recentBlockhash"] = recent_hash
+            return json.loads(
+                VersionedTransaction(
+                    message=MessageV0.from_json(json.dumps(transaction["message"][1])),
+                    keypairs=signers,
+                ).to_json()
+            )
```
Comment (angrybayblade) [inline plugins/aea-ledger-solana/aea_ledger_solana/crypto.py:188]: @Dassy23 Please check this logic for signing versioned txs, I ported this from the issue you opened

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py [hunk 3]
```diff
@@ -776,9 +776,17 @@ def deserialize_tx(
         if not isinstance(tx, Dict):
             raise ValueError(f"Unknown transction type found `{type(tx)}`")
 
+        if isinstance(tx["message"], list):
+            _, tx["message"] = tx["message"]
+            return SolanaTransaction.from_solders(
+                SoldersTransaction.from_json(json.dumps(tx))
+            )
```
[the remark this answers is not in the document]
Comment (Dassy23) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:783]: I don't think serializing to legacy will work. You can vTx -> json but on the reverse you have to pull out the messagev0 part and load into a new vTx and then serialize from there . Shouldn't need a keypair to serialize

File: plugins/aea-ledger-solana/aea_ledger_solana/solana.py [hunk 4]
```diff
@@ -778,8 +779,12 @@ def deserialize_tx(
 
         if isinstance(tx["message"], list):
             _, tx["message"] = tx["message"]
+            _, tx["signatures"] = tx["signatures"]
             return SolanaTransaction.from_solders(
-                SoldersTransaction.from_json(json.dumps(tx))
+                SoldersVersionedTransaction.populate(
+                    message=MessageV0.from_json(json.dumps(tx["message"])),
+                    signatures=[Signature.from_bytes(tx["signatures"])],
+                )
```
Comment (angrybayblade) [inline plugins/aea-ledger-solana/aea_ledger_solana/solana.py:787]: I think this should work @Dassy23

Comment (Dassy23) [reply plugins/aea-ledger-solana/aea_ledger_solana/solana.py:787]: yupp looks good

**Review by solarw — APPROVED** [review]
lgtm

**Pull request merged**